src/esn.calendar.libs/app/services/shells/calendar-shell.js

```diff
@@ -801,6 +802,48 @@ require('./valarm-shell.js');
       });
     }
 
+    /**
+     * Returns the differences between this and the object to compare
+     * @param  {CalendarShell} that an event
+     *
+     * @return {Array} an array containing the keys of differences between events.
+     */
+    function getChanges(that) {
+      const changes = [];
+      const self = this;
+
+      CAL_EVENT_MODIFY_COMPARE_KEYS.map(key => {
+        switch (key) {
+        case 'start':
+        case 'end':
+          // check if both are not undefined and one of them is undefined
+          if (!(!self[key] && !that[key]) && (!self[key] || !that[key])) {
+            changes.push(key);
+          } else if (self[key] !== that[key] && (self[key]._isAMomentObject && that[key]._isAMomentObject && !self[key].isSame(that[key]))) {
+            changes.push(key);
+          }
+          break;
+
+        case 'rrule':
+        case 'alarm':
+          if (!(!self[key] && !that[key])) {
+            if ((!self[key] || !that[key]) || !self[key].equals(that[key])) {
+              changes.push(key);
+            }
+          }
+          break;
+
+        default:
+          if (!angular.equals(self[key], that[key])) {
+            changes.push(key);
+          }
+          break;
+        }
+      });
```

Comment:
The code block is unreadable I think, you have `switch-case`, `if-else` and nested `if` block

Reply:
i rewrote the whole block using `if` and `if-else` statements.